client/my-sites/site-settings/delete-site/index.jsx

```diff
@@ -45,6 +45,27 @@ module.exports = React.createClass( {
 		this.props.sites.off( 'change', this._updateSite );
 	},
 
+	renderNotice: function() {
+		var site = this.state.site;
+
+		if ( ! site ) {
+			return null;
+		}
+
+		return (
+			<Notice status="is-warning" showDismiss={ false }>
+				{ this.translate( '{{strong}}%(domain)s{{/strong}} will be unavailable in the future.', {
+					components: {
+						strong: <strong />
+					},
+					args: {
+						domain: site.domain
+					}
+				} ) }
+			</Notice>
+		);
```

Comment:
I'm not sure if there is a benefit either way, but if it is arbitrary, my preference is for the way @Tug implemented it, because:

- It works with multiple conditions that bail out early (example from `ManagePurchase`):

```js
if ( ! isRenewable( purchase ) && ! isRedeemable( purchase ) ) {
	return null;
}

if ( ! isExpired( purchase ) ) {
	return null;
}
```

- It's easier (for me) to read JSX that isn't wrapped in an extra block/blocks.